e2e/_suites/kubernetes-autodiscover/features/filebeat.feature

```diff
@@ -35,3 +35,10 @@ Scenario: Logs collection from a pod with an ephemeral container
     And "filebeat" collects events with "kubernetes.container.name:container-in-pod"
    When an ephemeral container is started in "a pod"
    Then "filebeat" collects events with "kubernetes.container.name:ephemeral-container"
+
+# Not really autodiscover, but this is a frequent use case in autodiscover-like
+# scenarios for logs collection.
+Scenario: Enrichment of normal input using add_kubernetes_metadata
+  Given "filebeat" is running with "input and add_kubernetes_metadata"
```

Comment:
Sorry to jump-in late: what would happen if `"filebeat" is running with "input"` or `"filebeat" is running with "add_kubernetes_metadata"` (only of them)?

Reply:
I am including this specific case of having both an input + `add_kubernetes_metadata` because it is included as default in the reference manifests for filebeat ([here](https://github.com/elastic/beats/blob/877d8bcd176b2f5d4efd2a81846a481b94798b49/deploy/kubernetes/filebeat-kubernetes.yaml#L12-L20)), and is a pretty common use case in general.

If `"filebeat" is running with "input"`, without `add_kubernetes_metadata` and without any autodiscover, then this is not a kubernetes use case, and would be covered by other suites.

If `"filebeat" is running with "add_kubernetes_metadata"` but without any input, then this will be a noop, the processor won't do anything. We could add a case with `add_kubernetes_metadata`, without any input, but with autodiscover, but this is a case we don't recommend, because autodiscover already makes the same metadata enrichment that  `add_kubernetes_metadata` does, so more resources are used to do the same thing again.

Reply:
Ok, understood. Then if suggest creating the second scenario, as it seems a feasible use case. The scenario could explicitly explain what you describe

Reply:
PR to add this scenario: https://github.com/elastic/e2e-testing/pull/1385